src/main/java/io/vertx/core/cli/ArgumentModel.java

```diff
@@ -0,0 +1,238 @@
+/*
+ *  Copyright (c) 2011-2013 The original author or authors
+ *  ------------------------------------------------------
+ *  All rights reserved. This program and the accompanying materials
+ *  are made available under the terms of the Eclipse Public License v1.0
+ *  and Apache License v2.0 which accompanies this distribution.
+ *
+ *       The Eclipse Public License is available at
+ *       http://www.eclipse.org/legal/epl-v10.html
+ *
+ *       The Apache License v2.0 is available at
+ *       http://www.opensource.org/licenses/apache2.0.php
+ *
+ *  You may elect to redistribute this code under either of these licenses.
+ */
+
+package io.vertx.core.cli;
+
+import io.vertx.codegen.annotations.DataObject;
+import io.vertx.core.json.JsonObject;
+
+import java.util.Objects;
+
+/**
+ * Defines a command line argument. Unlike options, argument don't have names and are identified using an index. The
+ * first index is 0 (because we are in the computer world).
+ *
+ * @author Clement Escoffier <clement@apache.org>
+ * @see OptionModel
+ */
+@DataObject(generateConverter = true)
+public class ArgumentModel {
```

Comment:
Why not name this "Argument" instead of "ArgumentModel" ? besides TypedArgument that extends it is not named "TypeArgumentModel".